### ISSUE TITLE
[Telemetry] Add telemetry for Install Source

### ISSUE THREAD

**Issue by amedyne** [mozilla-mobile/fenix#22138]
Similar to what we did in Fennec with https://bugzilla.mozilla.org/show_bug.cgi?id=1221646

Using `PackageManager.getInstallerPackageName()` we can determine from which App Store Focus was installed. And likely also filter out installs that did not originate from Google Play.

It will give us values like this:

```
null - developer
com.android.vending - google play
com.amazon.venezia - amazon app
com.sec.android.app.samsungapps - samsung app store
```
See also this for Focus-Android as reference: https://github.com/mozilla-mobile/focus-android/issues/5684

**Comment by Mugurell** [comment]
@ QA: This may be a bit harder to test since it would involve trying different app stores.
In general we only need to know that the probe exists in the metrics ping. If it contains "com.android.vending" for an app installed from Playstore then it will contain the appropriate data when installed from other stores.

**Comment by AndiAJ** [comment]
Hi, verified as fixed on the following builds using a OnePlus A3 (Android 6.0.1)

✅  Manually installed Firefox 95.0.0 Beta 1 (Not available yet on Playstore) - Metrics Ping e1067643-e042-4185-9f5c-fe7dcddfaad7
`"metrics.install_source": ""`
[Logcat](https://github.com/mozilla-mobile/fenix/files/7468604/instasourceB.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/AJ-0311000)



✅  Installed the latest Nightly build from the Play Store  - Metrics Ping 215f4a68-b817-4fb0-855a-50689eacada8
`"metrics.install_source": "com.android.vending"`
[Logcat](https://github.com/mozilla-mobile/fenix/files/7468650/instN.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/AJ-0311001)
